rust/main/chains/hyperlane-sealevel/src/interchain_gas.rs

```diff
@@ -41,7 +41,12 @@ impl SealevelInterchainGasPaymaster {
         conf: &ConnectionConf,
         igp_account_locator: &ContractLocator<'_>,
     ) -> ChainResult<Self> {
-        let provider = SealevelProvider::new(rpc_client, igp_account_locator.domain.clone(), conf);
+        let provider = SealevelProvider::new(
+            rpc_client,
+            igp_account_locator.domain.clone(),
+            igp_account_locator.address,
```

Comment:
I guess another observation is the IGP and Message indexing will race one another when it comes to the tx's recipient. Iiuc each task calls this https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/main/agents/scraper/src/db/txn.rs#L72 somewhere along the way, but it doesn't update if the tx hash already exists. So txs that include an IGP and Mailbox call (most) will be in the db with either